File: src/main/java/com/example/ssauc/user/login/util/PasswordValidator.java
package com.example.ssauc.user.login.util;

public class PasswordValidator {

    /**
     * 예: 최소 8자, 영문/숫자/특수문자 1개 이상
     */
    public static boolean isValid(String password) {
        if (password == null) return false;
        return password.matches("^(?=.*[A-Za-z])(?=.*\\d)(?=.*[@!#$%^&*])[A-Za-z\\d@!#$%^&*]{8,}$");
    }
}

File: src/main/java/com/example/ssauc/user/login/util/PasswordValidator.kt
package com.example.ssauc.user.login.util

object PasswordValidator {
    /**
     * 예: 최소 8자, 영문/숫자/특수문자 1개 이상
     */
    @JvmStatic
    fun isValid(password: String?): Boolean {
        if (password == null) return false
        return password.matches("^(?=.*[A-Za-z])(?=.*\\d)(?=.*[@!#$%^&*])[A-Za-z\\d@!#$%^&*]{8,}$".toRegex())
    }
}
